### PR TITLE
[T2A2][W09-A2]Lee Pei Xuan

### PR DESCRIPTION
**Pull request by pyxyyy**

### CODE REVIEW

**Comment by pyxyyy** [comment]
Ready for review

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -1154,6 +1172,19 @@ private static String removePrefixSign(String s, String sign) {
...
+    */
+    /*
+    private static String removePrefix(String fullString, String prefix) { 
+        if (fullString.startsWith(prefix)) {
+            return fullString.replaceFirst(fullString, prefix); 
+        }
+        return fullString;
+    }
+    */
+
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/AddressBook.java:1187]: Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -601,11 +616,19 @@ private static void executeExitProgramRequest() {
     private static String getUserInput() {
         System.out.print(LINE_PREFIX + "Enter command: ");
         String inputLine = SCANNER.nextLine();
-        // silently consume all blank and comment lines
-        while (inputLine.trim().isEmpty() || inputLine.trim().charAt(0) == INPUT_COMMENT_MARKER) {
-            inputLine = SCANNER.nextLine();
+        return consumeBlankAndCommentLines(inputLine);
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/AddressBook.java:619]: Think whether making it like this violates SLAP or not. :)

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -601,11 +616,19 @@ private static void executeExitProgramRequest() {
...
+    }
+
+    /**
+     * Checks if input line is blank or a comment and consumes it if so.
+     */
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/AddressBook.java:624]: Don't forget to add `@param` and `@return`. Good job for adding header comment.

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -339,8 +350,10 @@ private static boolean hasValidParentDirectory(Path filePath) {
      * If a file already exists, it must be a regular file.
      */
     private static boolean hasValidFileName(Path filePath) {
-        return filePath.getFileName().toString().lastIndexOf('.') > 0
-                && (!Files.exists(filePath) || Files.isRegularFile(filePath));
+        boolean hasExtension = filePath.getFileName().toString().lastIndexOf('.') > 0
+        boolean isExisting = !Files.exists(filePath)
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/AddressBook.java:354]: No semi-colon? Did you even compile it? Hmm..

File: src/seedu/addressbook/AddressBook.java [hunk 5]
```diff
@@ -250,16 +255,23 @@ private static void echoUserCommand(String userCommand) {
     }
 
     /**
+     * Throws the desired error message and quits the program if invalid arguments are pased. 
+     */
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/AddressBook.java:259]: Don't forget to add `@param` and there is a typo in the header comment.

**Review by jeffryhartanto — COMMENTED** [review]
@pyxyyy Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**